mr/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -212,6 +212,14 @@ public static boolean hiveCatalog(Configuration conf, Properties props) {
       return Optional.empty();
     } else {
       String name = catalogName == null ? ICEBERG_DEFAULT_CATALOG_NAME : catalogName;
+
+      String catalogImpl = conf.get(InputFormatConfig.catalogPropertyConfigKey(
```

Comment:
I think instead of doing the check here it's simpler to do the check in `addCatalogPropertiesIfMissing` before we set `CatalogUtil.ICEBERG_CATALOG_TYPE` to something. By doing that, you do not need to get the actual impl like this and you can just check `catalogProperties.containsKey(CatalogProperties.CATALOG_IMPL)`. This also allows compatibility with legacy catalog impl Hadoop config key.

Reply:
good point! I will change this then and add one more test for legacy configs